File: JS/frequency-counter.js
function mostFrequent(arr) {
    const freq = new Map();
    let bestNum = null;
    let bestCount = 0;

    for (const num of arr) {
        const count = (freq.get(num) || 0) + 1;
        freq.set(num, count);

        if (
            count > bestCount ||
            (count === bestCount && num < bestNum)
        ) {
            bestCount = count;
            bestNum = num;
        }
    }

    return bestNum;
}

// --- Read input from stdin ---
const lines = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
const n = parseInt(lines[0]);
const arr = lines[1].trim().split(/\s+/).map(Number).slice(0, n);

console.log(mostFrequent(arr));

File: C++/frequency-counter.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    int n;
    cin >> n;
    long long *arr = new long long[n];
    for (int i = 0; i < n; ++i) cin >> arr[i];

    unordered_map<long long, int> freq;
    long long bestNum = 0;
    int bestCount = 0;

    for (int i = 0; i < n; ++i) {
        long long num = arr[i];
        int count = ++freq[num];
        if (count > bestCount || (count == bestCount && num < bestNum)) {
            bestCount = count;
            bestNum = num;
        }
    }

    cout << bestNum << endl;
    delete[] arr;
    return 0;
}